pkg/search/proxy/store/multi_cluster_cache_test.go

```diff
@@ -0,0 +1,506 @@
+				return
+			}
+			if err != nil {
+				return
+			}
+			if !reflect.DeepEqual(tt.want.object, obj) {
+				t.Errorf("Objects diff: %v", cmp.Diff(tt.want.object, obj))
+			}
+		})
+	}
+}
+
+func TestMultiClusterCache_List(t *testing.T) {
+	cluster1 := newCluster("cluster1")
+	cluster2 := newCluster("cluster2")
+	cluster1Client := fakedynamic.NewSimpleDynamicClient(scheme,
+		newUnstructuredObject(podGVK, "pod11", withResourceVersion("1001"), withLabel("app", "foo")),
+		newUnstructuredObject(podGVK, "pod12", withResourceVersion("1002")),
+		newUnstructuredObject(podGVK, "pod13", withResourceVersion("1003"), withLabel("app", "foo")),
+		newUnstructuredObject(podGVK, "pod14", withResourceVersion("1004")),
+		newUnstructuredObject(podGVK, "pod15", withResourceVersion("1005")),
+	)
+	cluster2Client := fakedynamic.NewSimpleDynamicClient(scheme,
+		newUnstructuredObject(podGVK, "pod21", withResourceVersion("2001"), withLabel("app", "foo")),
+		newUnstructuredObject(podGVK, "pod22", withResourceVersion("2002")),
+		newUnstructuredObject(podGVK, "pod23", withResourceVersion("2003"), withLabel("app", "foo")),
+		newUnstructuredObject(podGVK, "pod24", withResourceVersion("2004")),
+		newUnstructuredObject(podGVK, "pod25", withResourceVersion("2005")),
+	)
+
+	newClientFunc := func(cluster string) (dynamic.Interface, error) {
+		switch cluster {
+		case cluster1.Name:
+			return cluster1Client, nil
+		case cluster2.Name:
+			return cluster2Client, nil
+		}
+		return fakedynamic.NewSimpleDynamicClient(scheme), nil
+	}
+	cache := NewMultiClusterCache(newClientFunc, restMapper)
+	err := cache.UpdateCache(map[*clusterv1alpha1.Cluster][]schema.GroupVersionResource{
+		cluster1: {podGVR},
+		cluster2: {podGVR},
+	})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	type args struct {
+		gvr     schema.GroupVersionResource
+		options *metainternalversion.ListOptions
+	}
+	type want struct {
+		resourceVersion string
+		names           sets.String
+		errAssert       func(error) bool
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "list pods",
+			args: args{
+				gvr:     podGVR,
+				options: &metainternalversion.ListOptions{},
+			},
+			want: want{
+				// fakeDynamic returns list with resourceVersion=""
+				resourceVersion: buildMultiClusterRV("cluster1", "", "cluster2", ""),
+				names:           sets.NewString("pod11", "pod12", "pod13", "pod14", "pod15", "pod21", "pod22", "pod23", "pod24", "pod25"),
+				errAssert:       noError,
+			},
+		},
+		{
+			name: "list pods with labelSelector",
+			args: args{
+				gvr: podGVR,
+				options: &metainternalversion.ListOptions{
+					LabelSelector: mustParseLabelSelector("app=foo"),
+				},
+			},
+			want: want{
+				// fakeDynamic returns list with resourceVersion=""
+				resourceVersion: buildMultiClusterRV("cluster1", "", "cluster2", ""),
+				names:           sets.NewString("pod11", "pod13", "pod21", "pod23"),
+				errAssert:       noError,
+			},
+		},
+	}
```

Comment:
Do we need to add a case to test the list limit?

Reply:
`limit` option doest not work in `fakeClientset`. I will add this case in e2e test.
```go
func ExampleFakeClient_ListLimit() {
	newPod := func(name string) *corev1.Pod {
		p := &corev1.Pod{}
		p.Name = name
		p.Namespace = "default"
		return p
	}

	cluster1Client := fakedynamic.NewSimpleDynamicClient(scheme,
		newPod("pod1"),
		newPod("pod2"),
		newPod("pod3"),
		newPod("pod4"),
		newPod("pod5"),
	)
	list, _ := cluster1Client.Resource(corev1.SchemeGroupVersion.WithResource("pods")).List(context.TODO(), metav1.ListOptions{Limit: 2})
	fmt.Print(len(list.Items))
	// Output: 5
}
```

Reply:
Updated. All are fixed.

Reply:
Ok, thanks~